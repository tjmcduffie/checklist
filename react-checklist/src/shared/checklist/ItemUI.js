import classnames from 'classnames';
import ItemCheckbox from './ItemCheckboxUI';
import React from 'react';

import '../base/css/Button.css';
import '../base/css/Flex.css';
import './Item.css';

const ItemUI = ({description, isComplete, onChange, onRemove, uuid}) => {
  const descriptionClassName = isComplete ? 'item-description-complete' : 'item-description';
  return (
    <li className={classnames('item', 'flex-row')}>
      <ItemCheckbox
        className={classnames('flex-auto')}
        isComplete={isComplete}
        onChange={onChange}
        uuid={uuid}
      />
      <span className={classnames(descriptionClassName, 'flex-grow', 'flex-align-center')}>
        {description}
      </span>
      <button
        className={classnames('clear-button', 'monospaced-button', 'flex-shrink', 'flex-align-top')}
        onClick={_ => onRemove(uuid)}
        onKeyPress={
          e => ((e.key === " " || e.key === "Enter") && onRemove(uuid))
        }>
        delete
      </button>
    </li>
  );
}

export default ItemUI;
