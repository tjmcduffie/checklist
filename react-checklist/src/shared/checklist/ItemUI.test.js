import ItemUI from './ItemUI';
import React from 'react';
import ReactDOM from 'react-dom';
import {create} from 'react-test-renderer';

describe('ItemUI', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ItemUI description="test" isComplete={false} onChange={noop} onRemove={noop} uuid="asdfasdf" />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('renders predictably', () => {
    it('when complete', () => {
      const elem = create(
        <ItemUI description="one" isComplete={false} onChange={noop} onRemove={noop} uuid="asdfasdf" />
      );
      const tree = elem.toTree();
      expect(tree.rendered.rendered.length).toBe(3);
      expect(tree.rendered.rendered[0].props.isComplete).toBe(false);
      expect(elem.toJSON()).toMatchSnapshot();
    });

    it('when incomplete', () => {
      const elem = create(
        <ItemUI description="one" isComplete={true} onChange={noop} onRemove={noop} uuid="asdfasdf" />
      );
      const tree = elem.toTree();
      expect(tree.rendered.rendered.length).toBe(3);
      expect(tree.rendered.rendered[0].props.isComplete).toBe(true);
      expect(elem.toJSON()).toMatchSnapshot();
    });
  });
});
