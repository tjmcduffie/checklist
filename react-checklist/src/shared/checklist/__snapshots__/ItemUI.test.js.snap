// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemUI renders predictably when complete 1`] = `
<li
  className="item flex-row"
>
  <span
    className="flex-auto item-checkbox-ui"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
  />
  <input
    checked={false}
    className="item-checkbox-input"
    hidden={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="item-description flex-grow flex-align-center"
  >
    one
  </span>
  <button
    className="clear-button monospaced-button flex-shrink flex-align-top"
    onClick={[Function]}
    onKeyPress={[Function]}
  >
    delete
  </button>
</li>
`;

exports[`ItemUI renders predictably when incomplete 1`] = `
<li
  className="item flex-row"
>
  <span
    className="flex-auto item-checkbox-ui item-checkbox-ui-complete"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
  />
  <input
    checked={true}
    className="item-checkbox-input"
    hidden={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="item-description-complete flex-grow flex-align-center"
  >
    one
  </span>
  <button
    className="clear-button monospaced-button flex-shrink flex-align-top"
    onClick={[Function]}
    onKeyPress={[Function]}
  >
    delete
  </button>
</li>
`;
